### PR TITLE
chore(src): adjust examples to work with Camunda Cloud

### PR DESCRIPTION
**Pull request by pihme**
Changes related to https://github.com/camunda-cloud/camunda-cloud-documentation/issues/89

Change set:
- Adjusts Readme to explain variant with Camunda Cloud and with local deployment
- Adjusts the examples to work with either a Camunda Cloud cluster or with a local deployment

### CODE REVIEW

**Comment by pihme** [comment]
Merge should wait until new documentation is released. Otherwise, some links are not yet available.

File: src/example-1.go [hunk 1]
```diff
@@ -5,14 +5,40 @@ import (
...
+*/
 func main() {
+  gatewayAddr := os.Getenv("ZEEBE_ADDRESS")
+  plainText:= false
+
+  if (gatewayAddr == "") {
+    gatewayAddr = ZeebeAddr
+    plainText = true
+  }
+
```
Comment (miguelpires) [inline src/example-1.go]: The formatting is off and there shouldn't be parentheses around the condition in the `if. I think this should fix it
```suggestion
    gatewayAddr := os.Getenv("ZEEBE_ADDRESS")
    var plainText bool

    if gatewayAddr == "" {
        gatewayAddr = ZeebeAddr
        plainText = true
    }
```

File: src/example-2.go [hunk 2]
```diff
@@ -4,14 +4,40 @@ import (
...
+authentication or encryption enabled.
+*/
 func main() {
+  gatewayAddr := os.Getenv("ZEEBE_ADDRESS")
+  plainText:= false
+
+  if (gatewayAddr == "") {
+    gatewayAddr = ZeebeAddr
+    plainText = true
+  }
```
Comment (miguelpires) [inline src/example-2.go]: The formatting is off and there shouldn't be parentheses around the condition in the `if`.
```suggestion
    gatewayAddr := os.Getenv("ZEEBE_ADDRESS")
    var plainText bool

    if gatewayAddr == "" {
        gatewayAddr = ZeebeAddr
        plainText = true
    }
```

File: src/example-3.go [hunk 3]
```diff
@@ -4,14 +4,40 @@ import (
...
+authentication or encryption enabled.
+*/
 func main() {
+  gatewayAddr := os.Getenv("ZEEBE_ADDRESS")
+  plainText:= false
+
+  if (gatewayAddr == "") {
+    gatewayAddr = ZeebeAddr
+    plainText = true
+  }
```
Comment (miguelpires) [inline src/example-3.go]: Same
```suggestion
    gatewayAddr := os.Getenv("ZEEBE_ADDRESS")
    var plainText bool

    if gatewayAddr == "" {
        gatewayAddr = ZeebeAddr
        plainText = true
    }
```

File: src/example-4.go [hunk 4]
```diff
@@ -7,17 +7,44 @@ import (
...
+authentication or encryption enabled.
+*/
 func main() {
+  gatewayAddr := os.Getenv("ZEEBE_ADDRESS")
+  plainText:= false
+
+  if (gatewayAddr == "") {
+    gatewayAddr = ZeebeAddr
+    plainText = true
+  }
```
Comment (miguelpires) [inline src/example-4.go]: Same
```suggestion
    gatewayAddr := os.Getenv("ZEEBE_ADDRESS")
    var plainText bool

    if gatewayAddr == "" {
        gatewayAddr = ZeebeAddr
        plainText = true
    }
```

File: src/example-1.go [hunk 5]
```diff
@@ -5,14 +5,40 @@ import (
...
+When connecting to a cluster in Camunda Cloud, this application assumes that the following
+environment variables are set:
+
+ZEEBE_ADDRESS
+ZEEBE_CLIENT_ID
+ZEEBE_CLIENT_SECRET
+ZEEBE_AUTHORIZATION_SERVER_URL
+
+Hint: When you create client credentials in Camunda Cloud you have the option
+to download a file with above lines filled out for you.
```
Comment (miguelpires) [inline src/example-1.go]:
```suggestion
to download a file with the lines above filled out for you.
```
Or something else, but "with above lines" sounds weird to me.

File: src/example-2.go [hunk 6]
```diff
@@ -4,14 +4,40 @@ import (
...
+When connecting to a cluster in Camunda Cloud, this application assumes that the following
+environment variables are set:
+
+ZEEBE_ADDRESS
+ZEEBE_CLIENT_ID
+ZEEBE_CLIENT_SECRET
+ZEEBE_AUTHORIZATION_SERVER_URL
+
+Hint: When you create client credentials in Camunda Cloud you have the option
+to download a file with above lines filled out for you.
```
Comment (miguelpires) [inline src/example-2.go]:
```suggestion
to download a file with the lines above filled out for you.
```

File: src/example-3.go [hunk 7]
```diff
@@ -4,14 +4,40 @@ import (
...
+When connecting to a cluster in Camunda Cloud, this application assumes that the following
+environment variables are set:
+
+ZEEBE_ADDRESS
+ZEEBE_CLIENT_ID
+ZEEBE_CLIENT_SECRET
+ZEEBE_AUTHORIZATION_SERVER_URL
+
+Hint: When you create client credentials in Camunda Cloud you have the option
+to download a file with above lines filled out for you.
```
Comment (miguelpires) [inline src/example-3.go]:
```suggestion
to download a file with the lines above filled out for you.
```

File: src/example-4.go [hunk 8]
```diff
@@ -7,17 +7,44 @@ import (
...
+When connecting to a cluster in Camunda Cloud, this application assumes that the following
+environment variables are set:
+
+ZEEBE_ADDRESS
+ZEEBE_CLIENT_ID
+ZEEBE_CLIENT_SECRET
+ZEEBE_AUTHORIZATION_SERVER_URL
+
+Hint: When you create client credentials in Camunda Cloud you have the option
+to download a file with above lines filled out for you.
```
Comment (miguelpires) [inline src/example-4.go]:
```suggestion
to download a file with the lines above filled out for you.
```

**Review by miguelpires — CHANGES_REQUESTED** [review]
Just a few small issues with formatting and the new code. I added suggestions to make it easier to change but feel free to do something different if you want. 
Also, I noticed this example has the readyClose channel used by the worker to signal that it can be closed. This was added to the old Zeebe docs to fix a deadlock in the code but I got some feedback from two users who thought it was confusing. I opened an issue for it https://github.com/zeebe-io/zeebe/issues/5444 this is out of scope for this PR but I just thought I'd bring it up since you're touching these files

**Comment by pihme** [comment]
@MiguelPires Since indentation seems to be important I added an https://editorconfig.org/ file. Not 100% sure on the settings though. Please have a look whether the settings are compliant with Go style.

**Comment by miguelpires** [comment]
You can just use `gofmt` to format the code according to the language's standards. I should've done that instead of eyeballing it in the Github suggestions, my bad. Also, did you run the examples?

**Comment by pihme** [comment]
I did run the examples with a local server

**Comment by pihme** [comment]
Should we remove `.editorconfig` then if there is a dedicated tool in the Go ecosystem?

**Comment by miguelpires** [comment]
> Should we remove `.editorconfig` then if there is a dedicated tool in the Go ecosystem?

Sure, you can do that and also squash so it's ready to merge

**Review by miguelpires — APPROVED** [review]
~~bors r+~~ Force of habit

Looks good :+1:

**Pull request merged**